### PR TITLE
Add postgresql-evr extension to katello

### PR DESCRIPTION
**Pull request by ianballou**
Adds the postgresql-evr package to katello.

### CODE REVIEW

File: manifests/application.pp [hunk 1]
```diff
@@ -72,6 +73,10 @@
     config_file => "${foreman::plugin_config_dir}/katello.yaml",
   }
 
+  package { "${$scl_postgresql}-postgresql-evr":
```
Comment (ekohl) [inline manifests/application.pp:76]: You don't need this if it's a remote database. This also breaks on EL8. You should really pass a variable `$postgresql_evr_package` and determine the full name in `params.pp` where you consider the OS version.

Comment (ianballou) [reply manifests/application.pp:76]: This package is only build for EL7, so it looks like I will just do nothing if we're on EL8

**Comment by ianballou** [comment]
@ekohl I think I've addressed your comments.

**Comment by ianballou** [comment]
We have to bump the postgresql-evr version again.  I will update the version in this PR once that is up.

File: manifests/params.pp [hunk 2]
```diff
@@ -22,13 +22,19 @@
...
   Stdlib::HTTPSUrl $pulp_url = "https://${facts['fqdn']}/pulp/api/v2/",
   Stdlib::HTTPSUrl $crane_url = "https://${facts['fqdn']}:5000",
   Stdlib::Host $qpid_hostname = 'localhost',
   String[1] $candlepin_oauth_key = $katello::globals::candlepin_oauth_key,
   String[1] $candlepin_oauth_secret = $katello::globals::candlepin_oauth_secret,
   Stdlib::Host $candlepin_host = 'localhost',
-  Stdlib::HTTPSUrl $candlepin_url = "https://${candlepin_host}:8443/candlepin"
+  Stdlib::HTTPSUrl $candlepin_url = "https://${candlepin_host}:8443/candlepin",
+  String[1] $scl_postgresql = 'rh-postgresql12',
+  String[1] $postgresql_evr_package = "postgresql-evr-0.0.2-1.el${$::os['release']['major']}.x86_64"
```
Comment (jlsherrill) [inline manifests/params.pp:38]: i think we'd want just the package name here, and then use  :ensure => latest  in the package delcaration above (unless i'm completely off base here)

Comment (ianballou) [reply manifests/params.pp:38]: Can I have the shortened package name here and still pick the right release (el7 vs el8)?

Comment (jlsherrill) [reply manifests/params.pp:38]: yeah, its basically doing a 'yum install $PKG_NAME'  so as long as its doing   'rh-postgresql12-postgresql-evr' on el7 and 'postgresql-evr' on el8, i think we're good

Comment (ianballou) [reply manifests/params.pp:38]: makes sense, I'll update

**Comment by jlsherrill** [comment]
@ianballou do those failures make any sense to you?

**Comment by ianballou** [comment]
@jlsherrill the don't, I'm going to test again locally.

File: manifests/application.pp [hunk 3]
```diff
@@ -72,6 +75,12 @@
     config_file => "${foreman::plugin_config_dir}/katello.yaml",
   }
 
+  if $manage_db {
+    package { "${$scl_postgresql}-${postgresql_evr_package}":
```
Comment (ekohl) [inline manifests/application.pp:79]: This doesn't allow no SCL as we want on EL8

File: manifests/application.pp [hunk 4]
```diff
@@ -72,6 +75,12 @@
...
+      ensure => latest,
```
Comment (ekohl) [inline manifests/application.pp:80]: We always use `installed`. Upgrading via Puppet is generally not what you want.
```suggestion
      ensure => installed,
```

File: manifests/params.pp [hunk 5]
```diff
@@ -22,13 +22,19 @@
...
   Stdlib::HTTPSUrl $pulp_url = "https://${facts['fqdn']}/pulp/api/v2/",
   Stdlib::HTTPSUrl $crane_url = "https://${facts['fqdn']}:5000",
   Stdlib::Host $qpid_hostname = 'localhost',
   String[1] $candlepin_oauth_key = $katello::globals::candlepin_oauth_key,
   String[1] $candlepin_oauth_secret = $katello::globals::candlepin_oauth_secret,
   Stdlib::Host $candlepin_host = 'localhost',
-  Stdlib::HTTPSUrl $candlepin_url = "https://${candlepin_host}:8443/candlepin"
+  Stdlib::HTTPSUrl $candlepin_url = "https://${candlepin_host}:8443/candlepin",
+  String[1] $scl_postgresql = 'rh-postgresql12',
+  String[1] $postgresql_evr_package = 'postgresql-evr'
```
Comment (ekohl) [inline manifests/params.pp:38]: What you can do is the following in `globals.pp`
```puppet
if $facts['os']['release']['major'] == '7' {
  $postgresql_evr_package = 'rh-postgresql12-postgresql-evr'
} else {
  $postgresql_evr_package = 'postgresql-evr'
}
```

Then:
```suggestion
  String[1] $postgresql_evr_package = $katello::globals::postgresql_evr_package,
```

This gives you sane defaults and ready for EL8 but allows you to override the package name if you need to and you want to experiment (via custom-hiera for example). If we change the SCL, we'd do a major version bump and change the hardcoded default.

**Comment by ianballou** [comment]
I'm not sure why those warnings are still there, I'm not hitting that locally.

File: manifests/application.pp [hunk 6]
```diff
@@ -63,6 +63,9 @@
   $candlepin_event_queue = $katello::params::candlepin_event_queue
   $crane_url = $katello::params::crane_url
   $crane_ca_cert = $certs::katello_server_ca_cert
+  $scl_postgresql = $katello::params::scl_postgresql
+  $postgresql_evr_package = $katello::params::postgresql_evr_package
+  $manage_db = katello::candlepin::manage_db
```
Comment (ekohl) [inline manifests/application.pp:68]: This is incorrect. Katello and Foreman are separate databases.
```suggestion
  $manage_db = $foreman::db_manage
```

**Comment by ekohl** [comment]
https://github.com/theforeman/puppet-katello/commit/6f42d22e45d34ef1615f59aef4d0e2b565510b9f was merged but had a bug and didn't actually ran the tests. That was fixed in a voxpupuli-tests 1.2 release, but I'm going through the modules to correct it, but that takes a bit of time.

**Comment by ianballou** [comment]
@ekohl thanks for the review, I've pushed the changes.  Just waiting now to test them out.

**Comment by jlsherrill** [comment]
[test]

**Comment by jlsherrill** [comment]
this still isn't running, @ianballou maybe it needs a rebase?

File: manifests/application.pp [hunk 7]
```diff
@@ -72,6 +74,12 @@
     config_file => "${foreman::plugin_config_dir}/katello.yaml",
   }
 
+  if $manage_db {
+    package { "${postgresql_evr_package}":
```
Comment (ehelms) [inline manifests/application.pp:78]: No need for quotes here you can reference the variable

**Comment by ianballou** [comment]
Tests are passing now, is the PR looking all set?

**Pull request merged**